lib/tree.rb

```diff
@@ -1,38 +1,62 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+class Tree
+  attr_reader :height, :age, :apples, :alive
 
   def initialize
+    @height = 0
+    @age = 0
+    @apples = []
+    @alive = true
   end
 
   def age!
+    @age += 1
```

Comment:
The `attr_reader` at the top exposes `@age`, etc. by calling methods `age`.

This line is both reading and assigning age if you used `attr_accessor`, to define the setter methods too, this line could be `age += 1`.

I prefer depending on the methods as much as possible. Ruby will gladly return `nil` for a typoed instance variable, but raises an error for a typoed method name.